pkg/stub/handler.go

```diff
@@ -187,12 +189,23 @@ func (h *Handler) StoreCurrentValidConfig(condition v1alpha1.SamplesResourceCond
 		// 4.0 testing showed that we were getting empty ConfigMaps
 		cm = nil
 	}
+	//TODO our create default samplesresource logic attempls to create the config map,
+	// but for now we have redundancy here in case it still does not exists
+	// Is this valid ??
 	create := false
 	if cm == nil {
 		create = true
 		cm = &corev1.ConfigMap{}
 		cm.Name = v1alpha1.SamplesResourceName
+	}
+
+	if cm.Data == nil {
 		cm.Data = map[string]string{}
+	}
+
+	// if this is the initial processing, then store the values for the spec
+	// values that cannot change after the initial processing
+	if _, ok := cm.Data[initialprocessingkey]; !ok {
```

Comment:
why wouldn't we just always store these values?  if it hasn't changed, it's a no-op.

Reply:
I think I'm struggling with my paranoia around future changes subverting assumptions of the current flow

I'll remove `initialprocessingkey`